core/src/main/java/net/adoptopenjdk/icedteaweb/resources/ResourceTracker.java

```diff
@@ -412,33 +417,35 @@ private void wait(Resource... resources) throws InterruptedException {
      * @throws InterruptedException if another thread interrupted the wait
      */
     private boolean wait(Resource[] resources, long timeout, TimeUnit timeUnit) throws InterruptedException {
+        Assert.requireNonNull(timeUnit, "timeUnit");
         if (timeout <= 0) {
             throw new IllegalArgumentException("Timout must be bigger than 0");
         }
-        long startTime = System.nanoTime();
-        long nanoTimeout = timeUnit.toNanos(timeout);
+        if(resources == null || resources.length == 0) {
+            return true;
+        }
+        final long startTimeInNanos = System.nanoTime();
+        final long nanosTillTimeout = timeUnit.toNanos(timeout);
 
-        // save futures in list to allow parallel start of all resources
-        final List<Future<Resource>> futures = Stream.of(resources)
-                .map(ResourceHandler::new)
-                .map(ResourceHandler::putIntoCache)
+        final List<Future<Resource>> collectedDownloadFutures = Arrays.asList(resources).stream()
+                .map(r -> triggerDownloadFor(r))
                 .collect(Collectors.toList());
 
-        for (Future<Resource> future : futures) {
-            final long nanoSinceStartOfMethod = System.nanoTime() - startTime;
-            final long waitTime = nanoTimeout - nanoSinceStartOfMethod;
-
-            if (waitTime <= 0) {
-                return false;
-            }
-
-            try {
-                future.get(waitTime, TimeUnit.NANOSECONDS);
-            } catch (TimeoutException e) {
-                return false;
-            } catch (ExecutionException ignored) {
-            }
-        }
-        return true;
+        return collectedDownloadFutures.stream()
```

Comment:
Not 100% sure but I believe the logic has changed in this block...
Maybe extract a method from the `map` block and give it a good name.
This may help to clarify if the logic is still the same

Reply:
will have a look